rendermodules/dataimport/generate_mipmaps.py

```diff
@@ -98,17 +98,16 @@ def run(self):
             zvalues1 = range(self.args['zstart'], self.args['zend'] + 1)
             # extract only those z's that exist in the input stack
             zvalues = list(set(zvalues1).intersection(set(zvalues)))
-        except NameError:
+        except KeyError:
             try:
                 if self.args['z'] in zvalues:
                     zvalues = [self.args['z']]
```

Comment:
If you pick a z outside of this zvalues in the stack, does this generate mipmaps for the entire stack?

Reply:
The zvalues list should be empty in that case and hence the code should exit gracefully without generating any mipmaps

Reply:
Russel is right.. if you only put self.args['z'] and that z value isn't in the stack, then the if statement isn't called, and zvalues remains all the zvalues which isn't the right behavior.  I'll fix it with an else statement and add a test.

Reply:
ok got it.... I was referring to line 100...